pkg/scaffold/input/input.go

```diff
@@ -144,8 +144,9 @@ type File interface {
 	GetInput() (Input, error)
 }
 
-// Validate validates input
-type Validate interface {
+// RequiresValidation is a file that requires validation
+type RequiresValidation interface {
```

Comment:
I agreed with you. It is hurting concepts as a single responsibility and cohesion. 
However, why change the name of the interface? Could we keep it as it was?

```suggestion
type Validation interface {
```

Reply:
The interface as you can see is used to check if a `File` requires to be validated. Look how the test looks like with this name:

```golang
func (_ *Scaffold) validate(file input.File) error {
	if reqValFile, ok := file.(input.RequiresValidation); ok {
		return reqValFile.Validate()
	}

	return nil
}
```

I think that is really readable, without knowing what the interface does, you can understand that function pretty easy. That was the reason behind the new name.

Reply:
IHMO if I call a `Validation()` it is implicit that the code `RequiresValidation`.
So, I'd prefer the previous name in order to avoid to be prolix, I mean unnecessary verbose and/or redundant.  However, I leave it to the maintainers and you decided.

Reply:
IMO using `RequiresValidation` is slightly better than the old name `Validate`.